runtime/vm/BytecodeInterpreter.hpp

```diff
@@ -148,6 +148,11 @@ ffiCallWithSetJmpForUpcall(J9VMThread *currentThread, ffi_cif *cif, void *functi
 #else /* FFI_NATIVE_RAW_API */
 ffiCallWithSetJmpForUpcall(J9VMThread *currentThread, ffi_cif *cif, void *function, UDATA *returnStorage, void **values);
 #endif /* FFI_NATIVE_RAW_API */
+
+#if JAVA_SPEC_VERSION >= 22
+extern void * memcpyFromObjectArray(J9VMThread *currentThread, j9object_t arrayObject, jboolean *isCopy, jboolean ensureMem32);
+extern void memcpyToObjectArray(J9VMThread *currentThread, j9object_t arrayObject, void *elems, jint mode, jboolean ensureMem32);
+#endif /* JAVA_SPEC_VERSION >= 22 */
```

Comment:
These declarations should be in `vm_internal.h` (and should not be version-specific).

Reply:
As explained at https://github.com/eclipse-openj9/openj9/pull/19263#issuecomment-2033206544, the code we leveraged is originally in `oti/ArrayCopyHelpers.hpp`, in which case we moved the common code to `vm/ArrayCopyHelpers.cpp` to have them shared at both cases (in JNI and our code in the interpreter).

That being said, it will become invisible for `oti/ArrayCopyHelpers.hpp` to call the common code if we move the declarations to `vm_internal.h` (only used internally in the `vm` directory), which ends up with compilation failures to find the common code around in the `oti` directory. That's why we have to deal with these declarations in this way to address both cases.

Reply:
It's entirely reasonable that there's a better header file in which to declare those functions, but repeating the declarations everywhere they're used is not reasonable.

Reply:
Maybe the declarations belong in `j9protos.h`, but I don't see how this is an insurmountable problem.

Reply:
I will double-check the declarations in `vm_internal.h` in compilation and move them to `j9protos.h` to see how it goes.

Reply:
It surprised me that the compilation & tests passed without any issue after moving the declarations to `vm_internal.h` (I didn't save the previous compilation failure logs caused by these declarations so not quite sure whether I misplaced them somewhere). Anyway, it is working now with these declarations set in  `vm_internal.h`